react/modules/class-component/README.MD

```diff
@@ -0,0 +1,47 @@
+[React](https://github.com/rolling-scopes-school/tasks/tree/master/react) Class Components 🌟
+
+## Module Overview 📚
+
+This module introduces students to Class Components in React. It aims to provide an in-depth understanding of how to
+create and manage class components in a React application. The module covers a range of topics from understanding what
+class components are to implementing them in a React application, and managing state and lifecycle methods.
+
+## Learning Objectives 🎯
+
+Students will:
+
+- Understand what Class Components are in React.
+- Learn how to create Class Components, including defining a class, rendering JSX, and using props.
+- Explore how to manage state in Class Components.
+- Get familiar with the lifecycle methods in Class Components.
+
+## Approximate Module Completion Time ⏱️
+
+- **[3 hours]**
+
+## Theory 📖
+
+Students are encouraged to study the following resources:
+
+1. **Understanding Class Components:**
+    - [Component](https://react.dev/reference/react/Component) - [1 hour]
+    - [React Official Docs - Components and Props](https://legacy.reactjs.org/docs/components-and-props.html) - [15 minutes]
+    - [React Official Docs - State and Lifecycle](https://legacy.reactjs.org/docs/state-and-lifecycle.html) - [15 minutes]
+2. **Implementing Class Components:**
+    - [React Tutorial - Class Components](https://www.tutorialspoint.com/reactjs/reactjs_components.htm) - [30 minutes]
+    - [Understanding Constructors with React Components](https://www.digitalocean.com/community/tutorials/react-constructors-with-react-components) - [30 minutes]
+
+[//]: # (TODO Practice?)
+
+## Practice 💻
+
+- Complete the test "Class Components in React" in the RS-App > Auto Test.
+- Complete the task "Implement Class Component in a React Application".
+
+[//]: # (TODO есть доп материалы?)
+
+## Additional Resources 📘
+
+Expand your knowledge with these additional materials:
```

Comment:
We can add materials from the legacy docs as well as from the actual (I've added the most important ones for the begginer):
https://legacy.reactjs.org/docs/react-component.html
https://react.dev/learn/your-first-component
https://react.dev/learn/passing-props-to-a-component
https://legacy.reactjs.org/docs/state-and-lifecycle.html
https://react.dev/learn/responding-to-events
https://react.dev/learn/rendering-lists
https://react.dev/learn/sharing-state-between-components